src/Realm.C

```diff
@@ -886,6 +889,8 @@ Realm::setup_nodal_fields()
 #ifdef NALU_USES_HYPRE
     stk::mesh::put_field_on_mesh(*hypreGlobalId_, *parts[ipart], nullptr);
 #endif
+    stk::mesh::put_field_on_mesh(*tpetGlobalId_, *parts[ipart], nullptr);
+    stk::mesh::field_fill(std::numeric_limits<LinSys::GlobalOrdinal>::max(), *tpetGlobalId_);
```

Comment:
We'll have to do some follow-up work to avoid having multiple id fields. We've talked about using the same id field for hypre and tpetra.

Reply:
I will remove the useless std::unique as part of this PR, the other tasks are < 2 days of coding, assuming a working build, more like 1 day if all goes well. 
Moving tpetGlobalID and hypreGlobalID into a new unified type is tedious typing,but should only be a day to get to a PR.